maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/task/StringTripleSupport.scala

```
@@ -0,0 +1,40 @@
+//   Copyright 2014 Commonwealth Bank of Australia
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package au.com.cba.omnia.maestro.core.task
+
+import au.com.cba.omnia.maestro.core.data.Field
+import au.com.cba.omnia.maestro.core.codec.{Decode, Tag}
+import au.com.cba.omnia.maestro.core.thrift.scrooge.StringTriple
+
+trait StringTripleSupport {
+
+  implicit val StringTripleDecode: Decode[StringTriple] = for {
+    first  <- Decode.of[String]
+    second <- Decode.of[String]
+    third  <- Decode.of[String]
+  } yield StringTriple(first, second, third)
+
+  implicit val StringTripleTag: Tag[StringTriple] = {
+    val fields =
+      Field("FIRST",  (p:StringTriple) => p.first) +:
+      Field("SECOND", (p:StringTriple) => p.second) +:
+      Field("THIRD",  (p:StringTriple) => p.third) +:
+      Stream.continually[Field[StringTriple,String]](
```

Hold on, not so fast!

The code created by our tag macro silently truncates columns to fit the fields expected by tag, rather then throwing an error (and by truncating, stops decode from throwing an error later too). This is an existing bug in maestro, which IIRC we already have an issue for.

This custom implementation is a slight tweak on the existing code produced by the macro, extending it with an infinite list of fields so that it won't silently truncate columns but will instead keep them around (meaning the decoder throws a meaningful error later). Although looking at it now, I can't remember why I didn't just add code to check the number of columns and throw an exception if it is wrong ...

Anyway, I think fixing these custom implementation is a decent idea, but it's far more important to fix the code generated by our macros, which is both far worse and actually going into production.